packages/discord.js/src/structures/ApplicationCommand.js

```diff
@@ -396,6 +397,21 @@ class ApplicationCommand extends Base {
     }
     return true;
   }
+  
+  /**
+   * When concatenated with a string, this automatically returns the command's mention
+   * instead of the ApplicationCommand object.
+   * <info>This only works with chat input application commands.</info>
+   * @returns {string}
+   * @example
+   * // Logs: Command: </ping:123456789012345678>
+   * console.log(`Command: ${command}`);
+   */
+  toString() {
+    return this.type === ApplicationCommandType.ChatInput
+      ? chatInputApplicationCommandMention(this.name, this.id)
+      : super.toString();
+  }
```

Comment:
A few things:

- This looks like a semver-major change.
- This does not account for chat commands with subcommands or subcommand groups and subcommands.
- This doesn't work for context menu commands, yet the class may be constructed by them as well.

Reply:
Addressing these one at a time,

1. The only scenario where I see this as breaking is if the developer relies on `ApplicationCommand#toString()` returning `[Object object]` since both pre- and post-change return a string.  This seems like a very, *very* specific edge case, but is understandable nonetheless.
2. That was one of the main problems with not being able to use parameters for those options; as of now, Discord still highlights the mentions for commands with the formatting missing a required subcommand/group, but they are unclickable.  As such, whether or not it remains a "valid" mention in the form that the `toString()` change returns is debatable.
3. That's entirely correct.  By checking for the chat input type, this will still return a valid string for context menu commands, but it will remain inconsistent with the return value for chat input commands.

Overall, I see how this change feels inconsistent and much more iffy than the original intentions.  I have no qualms if this feels unfeasible and should be scrapped since the builder mention function serves the same purpose but with greater functionality/practicality.